src/stateFromElement.js

```diff
@@ -75,14 +75,17 @@ const ELEM_ATTR_MAP = {
 };
 
 const getEntityData = (tagName: string, element: DOMElement) => {
-  let data = {};
+  const data = {};
   if (ELEM_ATTR_MAP.hasOwnProperty(tagName)) {
-    let attrMap = ELEM_ATTR_MAP[tagName];
-    for (let attr of Object.keys(attrMap)) {
-      let dataKey = attrMap[attr];
-      let dataValue = element.getAttribute(attr);
-      if (dataValue != null) {
-        data[dataKey] = dataValue;
+    const attrMap = ELEM_ATTR_MAP[tagName];
+    for (let i = 0; i < element.attributes.length; i++) {
+      const attr = element.attributes[i];
+      let dataKey = attrMap[attr.name] || attr.name.match(/^data-([a-z0-9-]+)$/);
+      if (Array.isArray(dataKey)) {
```

Comment:
I prefer if we can do this without reflection (asking the runtime which type `dataKey` is). In this case `dataKey` can be either a string or an Array but we don't know until runtime. It makes it hard for the IDE and static analysis tools to give us actionable type hints. In this case we can easily branch based on the existence of `attr.name` in `attrMap` like you did in https://github.com/sstur/draft-js-export-html/pull/23. I like that approach.